### ISSUE TITLE
Errors extending schemas with yup >= 0.30

### ISSUE THREAD

**Issue by leonbloy** [ritchieanesco/json-schema-yup-transform#53]
It seems I cannot nest a yup scheme generated by this library inside a raw yup scheme.

Example: Let's start with 

```
const SCHEMA_AB = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  $id: 'http://example.com/person.schema.json',
  type: 'object',
  properties: {
    a: {      type: 'string',      maxLength: 3,    },
    b: {      type: 'string',      maxLength: 3,    },
  },
  required: ['a', 'b'],
};

 const yupSchemaAb = convertToYup(SCHEMA_AB);
 const yupSchemaAbRaw = yup.object().shape({ 
    a: yup.string().required().max(3),
    b: yup.string().required().max(3),
 });
```

In this case, the two schemas are semantically equivalent, and indeed they work as expected:

```
    // this works
    let res = null;
    res = await yupSchemaAbRaw.validate({ a: 'x', b: 'z' });
    expect(res).toEqual({ a: 'x', b: 'z' });
    res = await yupSchemaAb.validate({ a: 'x', b: 'z' });
    expect(res).toEqual({ a: 'x', b: 'z' });
    res = await expect(yupSchemaAbRaw.validate({ a: 'x', b: 'too long' })).rejects.toThrow(Error);
    res = await expect(yupSchemaAb.validate({ a: 'x', b: 'too long' })).rejects.toThrow(Error);

```

However, if I put those equivalent schemas inside another, the generated one does not work

```
    const yupSchemaFull = yup.object().shape({
      text: yup.string().required().max(5),
      ab: yupSchemaAb,
    });
    const yupSchemaFullRaw = yup.object().shape({
      text: yup.string().required().max(5),
      ab: yupSchemaAbRaw,
    });
    res = await yupSchemaFullRaw.validate({ text: 's', ab: { a: 'x', b: 'z' } });
    expect(res).toEqual({ text: 's', ab: { a: 'x', b: 'z' } });
    res = await expect(yupSchemaFullRaw.validate({ text: 's', ab: { a: 'x', b: 'too long' } })).rejects.toThrow(Error);
    // ANY OF THE FOLLOWING TWO FAIL
    res = await yupSchemaFull.validate({ text: 's', ab: { a: 'x', b: 'z' } }); 
    res = await expect(yupSchemaFull.validate({ text: 's', ab: { a: 'x', b: 'too long' } })).rejects.toThrow(Error)
```

Both tests using `yupSchemaFull` fail with  `thrown: "Exceeded timeout of 15000 ms for a test.`

Internally, they seem to do the correct validation (in the "too long" case, a ValidationError is thrown to the console) but it seems there are something wrong with the sync/async logic...

Issue #21, seems to be related
https://github.com/ritchieanesco/json-schema-yup-transform/issues/21#issuecomment-1005765677 
but, contrarily yo the comments there I don't see a difference with yup versions (the above results are with yup 0.31.1 - which I tried after having problems with the current 0.32.11)

I'm using  json-schema-yup-transform 1.6.3 with Node 14, webpack 5.

**Comment by leonbloy** [comment]
Additionally: if instead of using `validate` we use `validateSync` , the problematic statements also fail, differently:

```
    // the following does not throw error, but returns undefined 
    res = yupSchemaFull.validateSync({ text: 's', ab: { a: 'x', b: 'z' } });  
    // same: should throw exception, but returns undefined
    res = expect(() => yupSchemaFull.validateSync({ text: 's', ab: { a: 'x', b: 'too long' } })).toThrow(Error);
```

**Comment by leonbloy** [comment]
Update: it all works as expected if I revert yup to version "0.29.1" 

Update2: it works until yup 0.29.3 , it breaks at yup 0.30.0
(the culprit -wild guess- might be https://github.com/jquense/yup/commit/dcae108f6674bd3c7472e4444c2e836681ff7bfd )

**Comment by ritchieanesco** [comment]
@leonbloy v0.29.3 introduced breaking changes that are incompatible with this library.

**Comment by leonbloy** [comment]
> @leonbloy v0.29.3 introduced breaking changes that are incompatible with this library.

(You probably meant v 0.30.0 - released in Nov 2020)
Well, then what? If this library is incompatible with the latest yup releases (that should documented prominently), are there plans to fix that?

**Comment by ritchieanesco** [comment]
Added the yup version this package supports in README [here](068086ee8e121128ba00d846339a5cfb0db4649e).
I may look to rewrite this library to be forward compatible, but that requires time and frankly, this was written for a work project in which I am no longer involved.

Feel free to raise a PR to make it compatible or fork the code and make changes to suit your needs.

**Comment by ritchieanesco** [comment]
@leonbloy  [2.0.0-beta.0](https://github.com/ritchieanesco/json-schema-yup-transform/tree/v2.0.0-beta.0) has just been published that supports `yup v0.32.11`.

**Comment by leonbloy** [comment]
> 2.0.0-beta.0  has just been published that supports yup v0.32.11.

It's been 11 months since that beta. Meanwhile several fixes have been released to the main branch (which, it seems, were not merged to that beta branch).  Also, a few days ago yup has released version 1.0. Are there any plans to merge and release a non beta version, compatible with yup latest version?
